### PR TITLE
[Tests] Test PhysX joint effort application

### LINKED ISSUE #7601: [Bug][PhysX backend] Joint effort targets silently not applied on isaacsim 6.0.1.0 (Newton backend works)
### 🐛 Bug Report

Joint effort targets are **silently not applied** by the PhysX backend on the isaacsim 6.0.1.0 stack: a constant effort command produces **zero** cart motion. The Isaac Lab-side buffers and actuator bookkeeping are all correct — the effort reaches `root_view.set_dof_actuation_forces()` and is then ignored by the simulation. The same test on the **Newton (MJWarp) backend works correctly**, so effort-controlled tasks (e.g. `Isaac-Cartpole-Direct-v0`) are untrainable on PhysX while appearing to run normally.

### Environment

- **OS:** Windows 11 Home (10.0.26200)
- **Python:** 3.12 (standalone venv)
- **isaacsim:** 6.0.1.0 (pip)
- **isaaclab:** 6.1.14 (source checkout, commit `fff603eafc6b74264a5261cc0183d6a65390d78`, tag `v3.0.0-beta2.patch1`)
- **isaaclab_physx:** 1.1.3 (source, same checkout)
- **torch:** 2.11.0+cu128
- **GPU:** NVIDIA GeForce RTX 4060 Laptop GPU
- **skrl:** 2.1.0 (used for the training-impact test below)

### Steps to reproduce

Run the attached `physx_effort_repro.py` against the **official** `Isaac-Cartpole-Direct-v0` task (no custom code involved). It applies a constant `+0.5` action (= +50 N on the cart joint, well below the 400 N effort limit) for 120 env steps and prints the cart state:

```bash
# PhysX backend (default preset) — affected
python physx_effort_repro.py

# Newton backend — works
python physx_effort_repro.py presets=newton_mjwarp
```

Actual output (PhysX):

```
[REPRO] t=  0: cart_pos=+0.00000 cart_vel=+0.00000
[REPRO] t= 59: cart_pos=-0.00000 cart_vel=+0.00037
[REPRO] t=119: cart_pos=-0.00000 cart_vel=+0.00041
[REPRO] RESULT: cart DID NOT MOVE (PhysX effort targets are silently dropped)
```

With `presets=newton_mjwarp` the cart accelerates immediately (`cart_vel=+0.51` after the first step, up to `+6.2 m/s`).

### Diagnostics (all on the PhysX backend)

1. **Isaac Lab-side state is correct at the API boundary.** After stepping with the constant effort:
   - `articulation._joint_effort_target_sim == [50.0, 0.0]` (staging buffer written by the `update_targets` kernel)
   - `articulation.data.joint_effort_target == [50.0, 0.0]`
   - `articulation.data.applied_torque ≈ [49.997, 0.0]`

   So the effort flows correctly through `_apply_actuator_model()` (ImplicitActuator `compute()` is a passthrough) into `write_data_to_sim()` → `root_view.set_dof_actuation_forces(self._joint_effort_target_sim, ...)`. Nothing downstream acts on it.

2. **Solver-independent.** Setting `solver_type=0` (PGS) → cart still exactly at 0.0000.

3. **Not the external-forces timing flag.** Setting `enable_external_forces_every_iteration=True` (the flag from the "TGS solver … may cause noisy velocities" warning) → same result.

4. **No drive-mode setup in the new backend.** There are no `drive_mode` / `set_dof_drive*` references anywhere in `isaaclab_physx` — I suspect `set_dof_actuation_forces` is ignored for DOFs whose PhysX drive is not in force mode on this stack, or the call is otherwise a silent no-op in isaacsim 6.0.1.0's tensor API.

5. **Newton backend unaffected.** The Newton actuation path (`_apply_actuator_model_newton` → `w.joint_f_2d` → `set_dof_actuation_forces`) applies the same effort targets correctly, so the problem is specific to the PhysX staging path on this stack.

### Impact on RL training

This makes effort-controlled Direct RL environments **impossible to learn** while looking like a normal training run. With the official `Isaac-Cartpole-Direct-v0` + skrl PPO (official agent config, 4096 envs), the mean total reward plateaus at ~10 (mean episode ≈ 38 of 300 steps) and never improves, regardless of hyperparameters (learning rate schedules, entropy scales, batch sizes all tested — identical plateau). The identical task/config on the Newton preset converges to **~296/300 mean reward within ~1000 steps**. The pole visibly swings only under gravity in both cases; the cart never responds.

### Expected behavior

Per the assets documentation, joint effort targets are feed-forward efforts applied in addition to the implicit-drive PD terms; a constant +50 N on the cart joint should accelerate the cart along the rail (as it does on the Newton backend).

### Notes

- Reproduces headless and with 16/4096 envs, on both the official task and a freshly generated extension-template copy of the same env.
- I could not test older isaacsim releases for comparison (no installable Isaac Lab 6.x/5.x artifacts exist on PyPI — only `3.0.0b2`), but the same workflow trained fine on earlier stacks, so this reads like a regression in the isaacsim 6.0.1.0 / isaaclab_physx 1.1.3 combination.
- Happy to run additional diagnostics (e.g. raw `set_dof_actuation_forces` outside of Isaac Lab, drive-mode readback) on request.

<details>
<summary>📄 physx_effort_repro.py (full source)</summary>

```python
# Copyright (c) 2022-2026, The Isaac Lab Project Developers.
# All rights reserved.
#
# SPDX-License-Identifier: BSD-3-Clause
"""Minimal reproduction: joint effort targets are silently not applied on the PhysX backend.

Applies a constant +0.5 action (=> +50 N on the cart joint of Isaac-Cartpole-Direct-v0)
for 120 env steps and prints the cart state. On a healthy stack the cart accelerates;
on the affected stack the cart stays exactly at 0.

Usage (run from any directory with `isaaclab` importable):
    python physx_effort_repro.py                     # PhysX backend (affected)
    python physx_effort_repro.py presets=newton_mjwarp  # Newton backend (works)

The `presets=...` token is consumed by `resolve_task_config` (Isaac-Cartpole-Direct-v0
ships CartpolePhysicsCfg with `physx` / `newton_mjwarp` presets).
"""

import argparse
import sys

parser = argparse.ArgumentParser(description=__doc__)
parser.add_argument("--num_envs", type=int, default=16)
args_cli, _hydra_args = parser.parse_known_args()

import isaaclab_tasks  # noqa: F401
from isaaclab.app import AppLauncher
from isaaclab_tasks.utils import resolve_task_config

app_launcher = AppLauncher(args_cli)
simulation_app = app_launcher.app

import gymnasium as gym
import torch

TASK = "Isaac-Cartpole-Direct-v0"
env_cfg, _ = resolve_task_config(TASK, "skrl_cfg_entry_point")
env_cfg.scene.num_envs = args_cli.num_envs
env_cfg.headless = True
env = gym.make(TASK, cfg=env_cfg)
num_envs = env.unwrapped.num_envs
device = env.unwrapped.device

obs, _ = env.reset()
o = obs["policy"]
print(f"[REPRO] t=  0: cart_pos={o[0, 2]:+.5f} cart_vel={o[0, 3]:+.5f}")
for t in range(120):
    actions = torch.full((num_envs, 1), 0.5, device=device)  # +50 N on the cart joint
    obs, _, _, _, _ = env.step(actions)
    o = obs["policy"]
    if t in (59, 119):
        print(f"[REPRO] t={t:3d}: cart_pos={o[0, 2]:+.5f} cart_vel={o[0, 3]:+.5f}")

moved = abs(o[0, 2]) > 1e-3
print(f"[REPRO] RESULT: cart {'MOVED' if moved else 'DID NOT MOVE'} "
      f"({'PhysX effort targets are working' if moved else 'PhysX effort targets are silently dropped'})")
env.close()
simulation_app.close()

```

</details>

### PR DESCRIPTION
**Pull request by StafaH**
# Description

Extend the existing PhysX effort-limit integration test to verify that a staged joint effort reaches the solver and produces joint motion. The existing test already creates a single-joint articulation for both CPU and CUDA and for configured/USD-authored effort limits. It now neutralizes the implicit PD target, applies +10 N·m, advances one physics step, and verifies the joint accelerates.

Issue #7601 reported that Isaac Sim 6.0.x accepted the effort tensor but silently failed to apply it in the PhysX solver. The Isaac Lab-side staging buffers were already correct, so changing actuator bookkeeping or emulating joint forces with body wrenches would hide the simulator defect and alter articulation semantics. PR #7685 upgraded both `develop` and `release/3.0.0` to Isaac Sim 6.1, where the reported PhysX reproduction now moves correctly. This assertion covers the previously missing observable behavior so a tensor-backend regression cannot pass by updating buffers alone.

The behavioral check reuses all existing simulation contexts and adds only one physics step per existing case. Measured aggregate test-call overhead is approximately 0.05 seconds; it adds no new test cases or simulator setup/teardown cycles.

Fixes #7601

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Release backport

- [x]  Backport this pull request to the active release branch after it merges into `develop`

## Screenshots

Not applicable.

## Validation

- `uv run --extra isaacsim python -m pytest source/isaaclab_physx/test/assets/test_articulation.py -k test_setting_effort_limit_writes_to_solver -q --durations=0` (4 passed: CPU/CUDA × configured/USD effort limit)
- `uv run isaaclab -f`
- `uv run python tools/changelog/cli.py check <current-upstream-base>`
- Re-ran the issue's constant-effort Cartpole scenario with `physics=isaacsim_physx` on Isaac Sim 6.1; the cart responds immediately and continues moving under the command.

## Checklist

- [x] I have read and understood the contribution guidelines
- [x] I have run the pre-commit checks with `uv run isaaclab -f`
- [x] Documentation changes are not needed for test-only regression coverage
- [x] My changes generate no new warnings
- [x] I have added tests that prove the fix is effective
- [x] I have added a changelog fragment under `source/isaaclab_physx/changelog.d/`
- [x] My name already exists in `CONTRIBUTORS.md`

### CODE REVIEW

**Comment by StafaH** [comment]
run-ci

**Review by ooctipus — APPROVED** [review]
